### PR TITLE
Added __CUDACC__ guard to host-device definitions.

### PR DESCRIPTION
**Pull request by corbett5**

### CODE REVIEW

File: src/chai/ChaiMacros.hpp [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
 #include "chai/config.hpp"
 
-#if defined(CHAI_ENABLE_CUDA)
+#if defined(CHAI_ENABLE_CUDA) && defined(__CUDACC__)
```
Comment (davidbeckingsale) [inline src/chai/ChaiMacros.hpp]: Nice. Have you run into this as a problem somewhere?

Comment (corbett5) [reply src/chai/ChaiMacros.hpp]: Right now in geos. I tried to get around it by adding cuda as a dependency to everything but that led to other issues. We do something similar in axom:
https://github.com/LLNL/axom/blob/6d8f3cf2ff523678bf20a9df69c4dabc6b028c31/src/axom/core/Macros.hpp#L28

Comment (davidbeckingsale) [reply src/chai/ChaiMacros.hpp]: Makes sense to me, thanks 👍

**Comment by corbett5** [comment]
@davidbeckingsale since the Azure tests passed can you merge this?

**Pull request merged**